klient/client.go

```diff
@@ -38,6 +40,12 @@ type client struct {
 	resources *resources.Resources
 }
 
+// NewControllerRuntimeClient provides an instance of the Controller runtime client with
+// the provided rest config and custom runtime scheme.
+func NewControllerRuntimeClient(cfg *rest.Config, scheme *runtime.Scheme) (cr.Client, error) {
```

Comment:
I am not sure if this is needed.  What I do think is needed is a way to inject an existing Controller Runtime client to be used be used with Resources:

```go
func (r *Resources) SetCRClient(c cr.Client) {
  r.client = c
}
```
Or something like that

Reply:
@vladimirvivien  THis might break a few things if we are not careful possibly ?

```golang
type Resources struct {
	config *rest.Config
	scheme *runtime.Scheme
	client cr.Client
	namespace string
}
```

If we are to override the `Resources.client` should we allow the same for `Resources.config` if so, how do we enforce that? Document it and hope everyone will read and keep that in mind ? Might get a bit messy is it not ?

Reply:
I think we should not let someone set the `Resources.Client` directly. Instead, use the `klient/client.go` method to create a new resource for custom config or use the `GetControllerRuntimeClient` function from the `Resources` to access the already existing client. That way, we can keep this clean.